scripts/fund-address.ts

```diff
@@ -0,0 +1,147 @@
+}
+
+async function fundNative(chain, sender: string, receiver: string, targetBalanceStr: string) {
+  const senderBalance = await getNativeBalance(chain, sender)
+  const balance = await getNativeBalance(chain, receiver)
+  const targetBalanceNr = moveDecimalPoint(targetBalanceStr, Balance.DECIMALS)
+  const targetBalance = new NativeBalance(new BN(targetBalanceNr))
+  const diff = targetBalance.sub(balance)
+
+  // stop if sender doesn't have enough funds
+  if (senderBalance.lte(diff)) {
+    console.log(
+      `sender ${sender} has ${senderBalance.toFormattedString()}, not enough to fund ${diff.toFormattedString()} to receiver ${receiver}`
+    )
+    return
+  }
+
+  // stop if receiver has enough funds already
+  if (diff.lte(new NativeBalance(new BN(0)))) {
+    console.log(
+      `receiver ${receiver} has ${balance.toFormattedString()}, ${diff.toFormattedString()} more than defined target ${targetBalance.toFormattedString()}`
+    )
+    return
+  }
+
+  // fund the difference
+  console.log(
+    `transfer ${diff.toFormattedString()} from ${sender} to ${receiver} to top up ${balance.toFormattedString()}`
+  )
+  await chain.withdraw('HOPR', receiver, diff.toString())
+}
+
+async function main() {
+  const argv = yargs(process.argv.slice(2))
+    .option('environment', {
+      describe: 'environment which determines the network to run on',
+      demandOption: true,
+      type: 'string'
+    })
+    .option('address', {
+      describe: 'ETH address',
+      demandOption: true,
+      type: 'string'
+    })
+    .option('erc20', {
+      describe: 'whether to fund ERC20 token instead of native',
+      demandOption: false,
+      type: 'boolean',
+      default: false
+    })
+    .option('target', {
+      describe: 'the target balance up to which the account shall be funded',
+      demandOption: true,
+      type: 'string'
+    })
+    .parseSync()
+
+  const environment = PROTOCOL_CONFIG.environments[argv.environment]
+  if (!environment) {
+    console.error(`Cannot find environment ${environment}`)
+    process.exit(1)
+  }
+
+  if (!PRIVATE_KEY) {
+    console.error('Environment variable PRIVATE_KEY missing')
+    process.exit(1)
+  }
+
+  // instantiate chain object based on given environment and private key
+  const network = PROTOCOL_CONFIG.networks[environment.network_id]
+  const chainOptions = {
+    chainId: network.chain_id,
+    environment: argv.environment,
+    gasPrice: parseGasPrice(network.gas_price),
+    network: environment.network_id,
+    provider: expandVars(network.default_provider, process.env)
+  }
+  const privKey = utils.arrayify(PRIVATE_KEY)
+  const chain = await createChainWrapper(chainOptions, privKey)
+  const sender = chain.getPublicKey().toAddress().toString()
+
+  // maybe fund ERC20
+  if (argv.erc20) {
+    await fundERC20(chain, sender, argv.address, argv.target)
+    return
+  }
+  // by default we fund native token
+  await fundNative(chain, sender, argv.address, argv.target)
+}
+
+main()
```

Comment:
This can reject and silently fail, please see: https://stackoverflow.com/questions/46515764/how-can-i-use-async-await-at-the-top-level

Reply:
I changed this accordingly.

Reply:
Can't see any changes?

Reply:
https://github.com/hoprnet/hoprnet/pull/2936/commits/a2d22e8b3395362480bc9bd9154bce254bd76b0f

Reply:
Oh, it didn't show up. Fine now.